common/src/test/java/org/opensearch/ml/common/utils/StringUtilsTest.java

```diff
@@ -861,14 +861,21 @@ public void testValidateFields_InvalidCharacterSet() {
     @Test
     public void prepareJsonValue_returnsRawIfJson() {
         String json = "{\"key\": 123}";
-        String result = StringUtils.prepareJsonValue(json);
+        String result = StringUtils.prepareJsonValue(json, false);
         assertSame(json, result);  // branch where isJson(input)==true
     }
 
+    @Test
+    public void prepareJsonValue_returnEscapeJsonIfForce() {
+        String json = "{\"key\": 123}";
+        String result = StringUtils.prepareJsonValue(json, true);
+        assertEquals("{\\\"key\\\": 123}", result);
+    }
+
     @Test
     public void prepareJsonValue_escapesBadCharsOtherwise() {
         String input = "Tom & Jerry \"<script>";
-        String escaped = StringUtils.prepareJsonValue(input);
+        String escaped = StringUtils.prepareJsonValue(input, false);
```

Comment:
Something is misleading here, you are actually escaping the plain text, but passing "escape" parameter as false. May be rename "escape" parameter in the API definition, which should understand it is only about json string.

Reply:
make sense, i will  update  it